[docs] Fixup Config; Golden Gate naming in Debugging, Targets files (#365)
firesim/firesim: docs/Advanced-Usage/Generating-Different-Targets.rst

To generate a specific instance of a target, the build system leverages four Make variables:

1. ``TARGET_PROJECT``: this points the Makefile (`sim/Makefile`) at the right

>    target-specific Makefrag, which defines the generation and Golden Gate-level

Reviewer: In other places in the docs you left this as MIDAS-level software simulation. Can we leave this as MIDAS-level, since Golden[-]Gate-Level because really long? (and you've left the term Midas hanging around in places like Midas-examples...)

Author: I intended to leave ML, ML for now. I agree GG-level isn't good. 

I didn't change anything that would require a code change. The midasexamples will remain the midasexamples for now.

    class Override2 extends Config((site, here, up) => {...})
    // Specify by setting TARGET_CONFIG=Compound
    class Compound extends Config(new Override2 ++ new Override1 ++ new Base)
    // OR by setting TARGET_CONFIG=Override2_Override1_Base
    // Can specify undefined classes this way. ex: TARGET_CONFIG=Override2_Base

With this scheme, you don't need to define a Config class for every instance you
wish to generate. We use this scheme to specify FPGA frequencies (eg.
"BaseF1Config_F90MHz") in manager build recipes, but it's also very useful for doing
sweeping over a parameterization space. 

> **Note that the precedence of Configs decreases from left to right in a string**.  Appending a config to an existing one will only have an effect if it

Reviewer: ```suggestion
**Note that the precedence of Configs decreases from left to right in a string**.  Appending (as opposed to prepending) a config to an existing one will only have an effect if it
```

Author: I think this is unambiguous enough, given that you've said it in other places, and there's an example below. The docs are too wordy as it is.

sets a field not already set in higher precendence Configs.

For example, "BaseF1Config_F90MHz" is equivalent to
"BaseF1Config_F90MHz_F80MHz" as ``DesiredHostFrequency`` resolves to 90 MHz,

> but "F90MHz_BaseF1Config" is distinct "F80MHz_F90MHz_BaseF1Config" in

Reviewer: ```suggestion
but "F90MHz_BaseF1Config" is distinct from "F80MHz_F90MHz_BaseF1Config" in
```

> that ``DesiredHostFrequency`` resolves to 90 and 80 MHz respectively.

Reviewer: ```suggestion
that ``DesiredHostFrequency`` resolves to 90 MHz in the former and 80 MHz in the latter.
```
